script.js

```diff
@@ -1,4 +1,94 @@
+var getRandomHand = function () {
+  var randomDecimal = Math.random() * 3;
+  var randomInteger = Math.floor(randomDecimal); 
+   if (randomInteger == 0) {
+    return 'scissors';
+  }
+   if (randomInteger == 1) {
+    return 'paper';
+  }
+    return 'stone'; 
+  }; 
+ 
+var numWin = 0; 
+var numGames = 0; 
+var username = ''; 
+var mode = '';
+
+// main function 
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
-};
+
+// player name 
+if(!username){
+  if(!input ) {
+    return 'What is your your preferred name for the game?'; 
+  }
+    username = input; 
+    return `Hello ${username}! Hit the button again! `
+}
+
+// chooseing a mode 
+if(!mode){
+  if(!input ) {
+    return 'Which mode do you want to play? A. reverse B. Korean C. Comp vs Comp D. normal'; 
+  }
+    mode = input; 
+    var selection = function (){
+      if(input == 'A') {
+        return 'reverse'
+      }
+      if(input == 'B') {
+        return 'Korean'
+      }
+      if(input == 'C') {
+        return 'Comp vs Comp'
+      }
+      if(input == 'D') {
+        return 'normal'
+      }
+    };
+    var modename = selection(); 
+    return `You've chosen the ${modename} mode. <br><br>
+    Start the game by keying in any of the three words: scissors, paper, or stone.`; 
+}
+
+
+// userinput 
+  var playerinput = input; 
+ 
+  if(playerinput.substring(0,2) != 'sc' && playerinput.substring(0,2) != 'si' && playerinput.substring(0,2) != 'st' && playerinput.substring(0,2) != 'pa')
+  {
+  return `Remember to only choose among the three following words: scissors, paper, or stone. `
+  }
+
+
+
+
+  var robotinput = getRandomHand(); 
+  console.log(robotinput);
+  numGames += 1; 
+
+// normal mode  
+  var result = 'you lost ' + username + '! Your current winning rate is ' + (numWin/numGames).toFixed(2)*100 + '% of all the times';
+
+  if (robotinput == 'scissors' & playerinput.substring(0,2) == ('sc' || 'si')
```

Comment:
For complex logic like `playerinput.substring(0,2) == ('sc' || 'si')` that you repeat in multiple places, create a helper function for it. For example,

```
var evalIsScissors = function (input) {
  var prefix = input.substring(0, 2);
  return prefix == 'sc' || prefix == 'si';
}
```

Reply:
@jetljt your logic is broken. `playerinput.substring(0,2) == ('sc' || 'si')` does not translate to "does the input start with 'sc' or 'si'". This code actually checks if playerinput.substring(0,2) is equal to `true`, because `('sc' || 'si')` will evaluate to true.

If you want to check if input equals 'sc' or 'si', you'll have to write logic like `input == 'sc' || input == 'si'` instead.

Reply:
@jetljt your usage of the `&&` operator is also broken. You'll want to use `&&` instead of `&`. `&` refers to the bitwise and operator, which is something different that we do not want to do. Please pay attention to details!